courses/migrations/0018_backpopulate_courserun_expiration_date.py

```diff
@@ -0,0 +1,29 @@
+# Generated by Django 2.2.3 on 2019-08-06 10:14
+
+from datetime import timedelta
+
+from django.db import migrations
+
+
+def backpopulate_expiration_date(apps, schema_editor):
+    """
+    Fetch all CourseRuns with an end_date and without an expiration_date and set that field
+    to end_date + 90 days.
+    """
+    CourseRun = apps.get_model("courses", "CourseRun")
+    course_runs = CourseRun.objects.filter(
+        end_date__isnull=False, expiration_date__isnull=True
+    ).values("id", "end_date")
+    for course_run in course_runs:
+        CourseRun.objects.filter(id=course_run["id"]).update(
```

Comment:
Filtering second time can be omitted. I guess no need to get values in above query. You can do something like below
```
for course_run in course_runs:
    course_run.update(expiration_date=course_run["end_date"] + timedelta(days=90))
```
You can also look at https://docs.djangoproject.com/en/2.2/ref/models/querysets/#bulk-update and/or https://docs.djangoproject.com/en/2.1/topics/db/queries/#updating-multiple-objects-at-once

Reply:
Updated.